core/dbt/context/providers.py

```diff
@@ -1249,6 +1249,19 @@ def selected_resources(self) -> List[str]:
         """
         return selected_resources.SELECTED_RESOURCES
 
+    @contextmember
+    def submit_python_job(self, parsed_model: Dict, compiled_code: str) -> AdapterResponse:
+        # Check macro_stack and that the unique id is for a materialization macro
+        if not (
+            self.context_macro_stack.depth == 2
+            and self.context_macro_stack.call_stack[1] == "macro.dbt.statement"
+            and "materialization" in self.context_macro_stack.call_stack[0]
+        ):
+            raise RuntimeException(
+                f"submit_python_job is not intended to be called here, at model {parsed_model['alias']}, with macro call_stack {self.context_macro_stack.call_stack}."
+            )
+        return self.adapter.submit_python_job(parsed_model, compiled_code)
```

Comment:
QQ: Will `adapter.submit_python_job` still be callable (as a classmethod) from within the Jinja context? Would we need to decorate it as "unavailable"?

Reply:
It is no longer available from within Jinja context, we make functions available using `@available` decorator, and I removed that decorator for `submit_python_job` function. Also have a snowflake PR opened for it(given that snowflake actually submit things slightly different). Gonna also sync with the dbt-databricks maintainer to make sure they don't add that `@available` decorator anymore

Reply:
https://github.com/dbt-labs/dbt-core/pull/5822/files#diff-aa1e9e602a0f1e2930b465b30a35a28778fbe131da4fbf2c4e88dd52ca44f140L1223